packages/edit-post/src/components/layout/actions-panel.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * WordPress dependencies
+ */
+import { EntitiesSavedStates, PostPublishPanel } from '@wordpress/editor';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import PluginPostPublishPanel from '../sidebar/plugin-post-publish-panel';
+import PluginPrePublishPanel from '../sidebar/plugin-pre-publish-panel';
+
+export default function ActionsPanel() {
+	const {
+		closePublishSidebar,
+		togglePublishSidebar,
+		closeEntitiesSavedStates,
+		openEntitiesSavedStates,
+	} = useDispatch( 'core/edit-post' );
+	const {
+		publishSidebarOpened,
+		hasActiveMetaboxes,
+		isSaving,
+		isEntitiesSavedStatesOpen,
+		hasNonPostEntityChanges,
+	} = useSelect( ( select ) => {
+		return {
+			publishSidebarOpened: select(
+				'core/edit-post'
+			).isPublishSidebarOpened(),
+			hasActiveMetaboxes: select( 'core/edit-post' ).hasMetaBoxes(),
+			isSaving: select( 'core/edit-post' ).isSavingMetaBoxes(),
+			isEntitiesSavedStatesOpen: select(
+				'core/edit-post'
+			).isEntitiesSavedStatesOpen(),
+			hasNonPostEntityChanges: select(
+				'core/editor'
+			).hasNonPostEntityChanges(),
+		};
+	}, [] );
+
+	const openSavePanel = useCallback( () => openEntitiesSavedStates(), [] );
+
+	// It is ok for these components to be unmounted when not in visual use.
+	// So we can use them in a return cascade since only 1 is to be present at a time.
+	const Unmountables = () => {
```

Comment:
Ah good point.  

> A simple variable would suffice.

Do you mean to have a variable to use in the return statement with ternaries?  Or something else?

Do you think it would be suitable (if still necessary) to define this component standalone as opposed to inline inside the parent?

Reply:
Something like:

```jsx
export function Component( { x, y, z } ) {
	let content;
	if ( x ) content = <span>A</span>;
	else if ( y ) content = <span>B</span>;
	else if ( z ) content = <span>C</span>;
	return <div>Letter: { content }</div>;
}
```

Reply:
Thank you!  That makes sense here 😄